src/main/java/invtweaks/InvTweaks.java

```diff
@@ -448,16 +468,180 @@ int compareItems(@NotNull ItemStack i, @NotNull ItemStack j, int orderI, int ord
                     } else {
                         return jEnchs.size() - iEnchs.size();
                     }
+
                 } else {
-                    // TODO: It looks like Mojang changed the internal name type to ResourceLocation. Evaluate how much of a pain that will be.
-                    return ObjectUtils.compare(i.getItem().getRegistryName().toString(),
-                            j.getItem().getRegistryName().toString());
-                }
+                	//Two different items with the same sort number.  (Dictionary Items, Class Items, mergePrevious, or Unsorted Items.)
+
+                	//Allow external sorting systems to take control of unsorted items not handled by the tree.
+                	if (orderI == Integer.MAX_VALUE && api == true)
+                		return 0;
+
+                	Item iItem = i.getItem(), jItem = j.getItem();
+                	if (iItem != null && jItem != null) {
+
+                		//Sort By Harvest Level, (Better first.)
+                		int cTool = compareTools(i, j, iItem, jItem);
+                		if (cTool != 0) 
+                			return cTool;
+	            		
+	            		//Sort by main-hand damage capability:  (Higher first, faster first for same damage)
+                		int cSword = compareSword(i, j, iItem, jItem);
+                		if (cSword != 0) 
+                			return cSword;
+                		
+	                	//Sort By Armor utility:  (More First)
+                    	int cArmor = compareArmor(iItem, jItem);
+                    	if (cArmor != 0) 
+                    		return cArmor;
+                	}
+            	}
+            		
+        		//Use durability to sort, favoring more durable items.
+        		int maxDamage1 = i.getMaxDamage() <= 0 ? Integer.MAX_VALUE : i.getMaxDamage();
+        		int maxDamage2 = j.getMaxDamage() <= 0 ? Integer.MAX_VALUE : j.getMaxDamage();
+        		if (maxDamage2 != maxDamage1)
+        			return maxDamage2 - maxDamage1;  
+
+        		//Use remaining durability to sort, favoring more damaged.
+        		int curDamage1 = i.getItemDamage();
+        		int curDamage2 = j.getItemDamage();
+        		if (curDamage2 != curDamage1) {
+                    if(i.isItemStackDamageable() && !getConfigManager().getConfig().getProperty(InvTweaksConfig.PROP_INVERT_TOOL_DAMAGE).equals(InvTweaksConfig.VALUE_TRUE)) {
```

Comment:
There seem to be some general formatting inconsistencies around all this, especially in the preview this line is indented *less* than the one before it.

Reply:
I've been working on JEI and it is all-tabs-all-the-time.  It seems InvTweaks is all-space-all-the-time.  So, I'll need to keep that in mind and work out a fix.